src/optimizer.cpp

```diff
@@ -44,112 +47,181 @@ void Optimizer::getParams()
   auto getParam = utils::getParamGetter(node, name_);
   auto getParentParam = utils::getParamGetter(node, "");
 
-  getParam(model_dt_, "model_dt", 0.1);
-  getParam(time_steps_, "time_steps", 15);
-  getParam(batch_size_, "batch_size", 200);
-  getParam(iteration_count_, "iteration_count", 2);
-  getParam(temperature_, "temperature", 0.25);
-
-  getParam(base_constraints_.vx, "vx_max", 0.5);
-  getParam(base_constraints_.vy, "vy_max", 1.3);
-  getParam(base_constraints_.wz, "wz_max", 1.3);
-  constraints_ = base_constraints_;
+  auto & s = settings_;
+  getParam(s.model_dt_, "model_dt", 0.1);
+  getParam(s.time_steps_, "time_steps", 12);
+  getParam(s.batch_size_, "batch_size", 400);
+  getParam(s.iteration_count_, "iteration_count", 2);
+  getParam(s.temperature_, "temperature", 0.25);
+  getParam(s.base_constraints_.vx, "vx_max", 0.5);
+  getParam(s.base_constraints_.vy, "vy_max", 0.5);
+  getParam(s.base_constraints_.wz, "wz_max", 1.3);
+  getParam(s.sampling_std_.vx, "vx_std", 0.2);
+  getParam(s.sampling_std_.vy, "vy_std", 0.2);
+  getParam(s.sampling_std_.wz, "wz_std", 1.0);
+  s.constraints_ = s.base_constraints_;
 
-  getParam(sampling_std_.vx, "vx_std", 0.1);
-  getParam(sampling_std_.vy, "vy_std", 0.1);
-  getParam(sampling_std_.wz, "wz_std", 0.3);
   getParentParam(controller_frequency_, "controller_frequency", 0.0);
 
   std::string motion_model_name;
   getParam(motion_model_name, "motion_model", std::string("DiffDrive"));
   setMotionModel(motion_model_name);
 }
 
+// FIXME(artofnothingness) optimize this
+rcl_interfaces::msg::SetParametersResult Optimizer::dynamicParametersCallback(
```

Comment:
https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/smac_planner_hybrid.cpp#L376-L544 this is a decent example of how to do this in line with other examples.

Note: the "reinit" logic probably isn't needed here as much (or at all) so its really the rest I'm referencing. Here's a simpler example with just parameters of double type; https://github.com/ros-planning/navigation2/blob/main/nav2_planner/src/planner_server.cpp#L584-L612

Reply:
Though what you have now isn't unreasonable, I'm just noting because you have the note

Reply:
I'm in progress to make general param handler

Reply:
almost done, synchronization issues left